src/vs/vscode.d.ts

```diff
@@ -3429,6 +3460,15 @@ declare namespace vscode {
 		 * @return A new status bar item.
 		 */
 		export function createStatusBarItem(alignment?: StatusBarAlignment, priority?: number): StatusBarItem;
+
+		/**
+		 * Creates a [Terminal](#Terminal).
+		 *
+		 * @param name The optional name of a terminal, this will override the label used in the
+		 * terminal dropdown.
+		 * @return A new Terminal.
+		 */
+		export function createTerminal(name?: string): Thenable<vscode.Terminal>;
```

Comment:
To clarify - making the create-call sync doesn't mean the logic behind it must be all sync. Since all functions of the terminal are async under the hood anyways (talking to the main thread) you can just make them all wait on the initial creation (promise).